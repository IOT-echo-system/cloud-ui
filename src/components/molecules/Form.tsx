import {Stack, Typography} from '@mui/material'
import type {FormEvent} from 'react'
import React from 'react'
import type {FormInputType} from '../atoms'
import {Button, FormContainer, FormInput} from '../atoms'

export type FormProps = {
  handleSubmit: (event: FormEvent<HTMLFormElement>) => void
  title: string
  inputFields: FormInputType[]
  submitBtnText: string
  submitBtnDisabled?: boolean
  loading?: boolean
}

export const Form: React.FC<FormProps> = props => {
  return (
    <FormContainer onSubmit={props.handleSubmit}>
      <Stack spacing={2}>
        <Typography variant={'h5'}>{props.title}</Typography>
        {props.inputFields.map((inputField, index) => (
          <FormInput key={`input-${index}`} {...inputField} />
        ))}
        <Button
          loading={props.loading ?? false}
          type={'submit'}
          variant={'contained'}
          size={'large'}
          disabled={props.submitBtnDisabled}
        >
          {props.submitBtnText}
        </Button>
      </Stack>
    </FormContainer>
  )
}
