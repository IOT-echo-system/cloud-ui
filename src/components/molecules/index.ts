export * from './Header/Header'
export * from './Footer'
export * from './Form'
export * from './ValidatedProfile'
export * from './FetchDetails'
export * from './CustomModal'
export * from './BoardSecretKey/BoardSecretKey'
export * from './ConfirmationModals'
export * from './MqttSubscriber'
