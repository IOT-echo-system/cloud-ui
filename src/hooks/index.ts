export {default as useForm} from './useForm'
export * from './useForm'

export * from './useStore'
export * from './useCountDownTimer'
export {default as useCountDownTimer} from './useCountDownTimer'
