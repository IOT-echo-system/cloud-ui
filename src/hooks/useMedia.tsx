import {useMediaQuery} from '@mui/material'
import theme from '../theme/light'

export const useMedia = (): {[P in 'sm' | 'md' | 'lg' | 'xl' | 'xxl']: boolean} => {
  return {
    sm: useMediaQuery(theme.breakpoints.up('xs')),
    md: useMediaQuery(theme.breakpoints.up('sm')),
    lg: useMediaQuery(theme.breakpoints.up('md')),
    xl: useMediaQuery(theme.breakpoints.up('lg')),
    xxl: useMediaQuery(theme.breakpoints.up('xl'))
  }
}
