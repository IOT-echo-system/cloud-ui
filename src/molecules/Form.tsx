import {Stack, Typography} from '@mui/material'
import type {FormEvent} from 'react'
import React from 'react'
import type {FormInputType} from '@/atoms'
import {Button, FormContainer, FormInput} from '@/atoms'

export type FormProps = {
  handleSubmit: (event: FormEvent<HTMLFormElement>) => void
  title: string
  error?: string
  inputFields: FormInputType[]
  submitBtnText: string
  submitBtnDisabled?: boolean
}

export const Form: React.FC<FormProps> = props => {
  return (
    <FormContainer onSubmit={props.handleSubmit}>
      <Stack spacing={2}>
        <Typography variant={'h5'}>{props.title}</Typography>
        {props.error && (
          <Typography variant={'body1'} color={'error'}>
            {props.error}
          </Typography>
        )}
        {props.inputFields.map((inputField, index) => (
          <FormInput key={`input-${index}`} {...inputField} />
        ))}
        <Button type={'submit'} variant={'contained'} size={'large'} disabled={props.submitBtnDisabled}>
          {props.submitBtnText}
        </Button>
      </Stack>
    </FormContainer>
  )
}
