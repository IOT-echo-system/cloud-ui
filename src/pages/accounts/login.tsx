import type {NextPage} from 'next'
import React from 'react'
import {LogIn} from '@/templates/accounts'

const LoginPage: NextPage = () => {
  return <LogIn />
}

export default LoginPage
