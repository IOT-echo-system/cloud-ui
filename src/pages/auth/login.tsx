import type {NextPage} from 'next'
import React, {useEffect} from 'react'
import {LogIn} from '../../components/templates/auth'
import {AuthService} from '../../services'
import {Config} from '../../config'
import {useRouter} from 'next/router'

const LoginPage: NextPage = () => {
  const router = useRouter()
  useEffect(() => {
    AuthService.validate()
      .then(res => router.push(res.projectId ? Config.HOME_PAGE_PATH : Config.START_PAGE_PATH))
      .catch(() => ({}))
  }, [])

  return <LogIn />
}

export default LoginPage
