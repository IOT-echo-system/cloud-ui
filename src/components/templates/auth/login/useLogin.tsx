import {useRouter} from 'next/router'
import type React from 'react'
import type {ChangeEvent} from 'react'
import {useState} from 'react'
import type {FormInputType} from '../../../atoms'
import {useForm, useToast} from '../../../../hooks'
import {AuthService} from '../../../../services'
import {Config} from '../../../../config'

type UseLoginReturnType = {
  inputFields: FormInputType[]
  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void
  loading: boolean
}
const useLogin = (): UseLoginReturnType => {
  const router = useRouter()
  const {values, onChange, handleSubmit} = useForm({email: '', password: ''})
  const toast = useToast()
  const [loading, setLoading] = useState(false)

  const handleChange = <K extends keyof typeof values>(keyName: K) => {
    return (event: ChangeEvent<HTMLInputElement>) => {
      onChange(keyName, event.target.value)
    }
  }

  const onSubmit = () => {
    setLoading(true)
    AuthService.login(values)
      .then(() => router.push(Config.HOME_PAGE_PATH))
      .catch(toast.error)
      .finally(() => {
        setLoading(false)
      })
  }

  const inputFields: FormInputType[] = [
    {
      inputType: 'textField',
      type: 'email',
      value: values.email,
      onChange: handleChange('email'),
      label: 'Email',
      required: true
    },
    {
      inputType: 'textField',
      type: 'password',
      value: values.password,
      onChange: handleChange('password'),
      label: 'Password',
      required: true
    }
  ]

  return {
    handleSubmit: handleSubmit(onSubmit),
    inputFields,
    loading
  }
}

export {useLogin}
