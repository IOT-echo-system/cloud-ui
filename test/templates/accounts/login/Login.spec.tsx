import React from 'react'
import {render} from '@testing-library/react'
import * as UseLoginHooks from '../../../../src/templates/accounts/login/useLogin'
import {LogIn} from '../../../../src/templates/accounts'
import {Form} from '../../../../src/molecules/Form'
import * as MediaHook from '../../../../src/hooks/useMedia'

jest.mock('../../../../src/molecules/Form')
jest.mock('../../../../src/templates/accounts/login/useLogin')
jest.mock('../../../../src/hooks/useMedia')

describe('Login Template Test', () => {
  const mockUseLogin = {inputFields: [], error: '', handleSubmit: jest.fn()}

  beforeEach(() => {
    jest.resetAllMocks()

    jest.spyOn(UseLoginHooks, 'useLogin').mockReturnValue(mockUseLogin)
  })
  afterEach(jest.resetAllMocks)

  it('should match with the snapshot', () => {
    jest.spyOn(MediaHook, 'useMedia').mockReturnValue({md: true, sm: true, xl: false, xxl: false, lg: false})

    const {container} = render(<LogIn />)

    expect(UseLoginHooks.useLogin).toHaveBeenCalledTimes(1)
    expect(container).toMatchSnapshot()
    expect(MediaHook.useMedia).toBeCalledTimes(1)
    expect(Form).toHaveBeenCalledTimes(1)
    expect(Form).toHaveBeenCalledWith(
      {
        error: '',
        handleSubmit: expect.any(Function),
        inputFields: [],
        submitBtnText: 'Login',
        title: 'Login'
      },
      {}
    )
  })

  it('should match with the snapshot for mobile', () => {
    jest.spyOn(MediaHook, 'useMedia').mockReturnValue({md: false, sm: true, xl: false, xxl: false, lg: false})

    const {container} = render(<LogIn />)

    expect(UseLoginHooks.useLogin).toHaveBeenCalledTimes(1)
    expect(container).toMatchSnapshot()
    expect(MediaHook.useMedia).toBeCalledTimes(1)
    expect(Form).toHaveBeenCalledTimes(1)
    expect(Form).toHaveBeenCalledWith(
      {
        error: '',
        handleSubmit: expect.any(Function),
        inputFields: [],
        submitBtnText: 'Login',
        title: 'Login'
      },
      {}
    )
  })
})
