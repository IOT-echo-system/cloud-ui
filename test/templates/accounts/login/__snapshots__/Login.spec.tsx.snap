// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Template Test should match with the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1edlbwa-MuiStack-root"
  />
</div>
`;
